docs/pubsub-api.rst

```diff
@@ -0,0 +1,255 @@
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> [topic.name for topic in pubsub.list_topics(project_id="my.project")]
+   ['topic_name']
+
+Delete a topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.delete()
+
+
+Publish messages to a topic
+---------------------------
+
+Publish a single message to a topic, without attributes:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.publish('this is the message_payload')
+   <message_id>
+
+Publish a single message to a topic, with attributes:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> topic.publish('this is another message_payload',
+   ...               attr1='value1', attr2='value2')
+   <message_id>
+
+Publish a set of messages to a topic (as a single request):
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> with topic:
+   ...     topic.publish('this is the first message_payload')
+   ...     topic.publish('this is the second message_payload',
+   ...                   attr1='value1', attr2='value2')
+   [<message_id1>, <message_id2>]
+
+
+Manage subscriptions to topics
+------------------------------
+
+Create a new pull subscription for a topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.create_subscription('subscription_name')
+
+Create a new pull subscription for a topic with a non-default ACK deadline:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.create_subscription('subscription_name',
+   ...                                          ack_deadline=90)
+
+Create a new push subscription for a topic:
+
+.. doctest::
+
+   >>> ENDPOINT = 'https://example.com/hook'
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.create_subscription('subscription_name',
+   ...                                          push_endpoint=ENDPOINT)
+
+Get an extant subscription for a topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('subscription_name')
+
+Attempt to get a non-extant subscription for a topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('nonesuch')
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in ?
+   NotFound: ...
+
+Update the ACK deadline for a subscription:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('subscription_name')
+   >>> subscription.modify_ack_deadline(90)
```

Comment:
Wait, so you're saying it's not possible to create a subscription with no `ack_deadline` and then update the `ack_deadline` ?

[The docs](https://cloud.google.com/pubsub/reference/rest/v1beta2/projects/subscriptions/modifyAckDeadline) are *super* misleading about this.

If I create a `subscription` that takes an `ackDeadlineSeconds` property and an API method called `modifyAckDeadline` on the `subscription` object -- I expect the `modifyAckDeadline` to ... change the one I provided.

In short, I'd expect...

- `POST projects/my-project/subscriptions/my-subscription:modifyAckDeadline` to change the ack deadline on the **subscription**.

- `POST projects/my-project/messages/message-ack-id-here:modifyAckDeadline` to change the ack deadline on the **message**.

The docs say that this message takes an `ackId` parameter, meaning it does as you say and modifies the ack deadline for JUST a single message. Was this an oversight?